packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/Controls.tsx

```diff
@@ -88,16 +96,57 @@ const LocationDetailEmptyMessage = () => {
 
 // @ts-expect-error TODO: add Controls assignment
 export const LocationDetailViewControls: LocationDetailViewControls = () => {
+  const [{ data, isLoading }, handleList] = useAction({
+    type: 'LIST_LOCATION_ITEMS',
+  });
+
+  const { result, nextToken } = data;
+  const resultCount = result.length;
+  const hasNextToken = !!nextToken;
+  const [{ path }] = useControl({ type: 'NAVIGATE' });
+
+  React.useEffect(() => {
+    handleList({
+      prefix: path,
+      options: { ...DEFAULT_LIST_OPTIONS, refresh: true },
+    });
+  }, [path, handleList]);
+
+  const onPaginateNext = () =>
+    handleList({
+      prefix: path,
+      options: { ...DEFAULT_LIST_OPTIONS, nextToken },
+    });
+
+  const { currentPage, handlePaginateNext, handlePaginatePrevious } =
```

Comment:
Will need to provide `disableRefresh` and `handleReset` to the `RefreshControl`

Reply:
Thanks, updated in latest to pass those as props.